tests/dataplicity/m2m/wsclient/test_wsclient.py

```diff
@@ -0,0 +1,39 @@
+import threading
+import time
+
+from dataplicity.m2m.wsclient import WSClient
+from websocket_server import wsserver_factory
+from ws4py.websocket import EchoWebSocket
+
+# this factory yields a fixture which we can use inside of testing functions.
+# it runs a websocket server on a separate thread, and will shutdown once the
+# test function returns.
+wsserver = wsserver_factory(EchoWebSocket)
+
+
+class WSClientThread(threading.Thread):
```

Comment:
I was hoping to push the tests for WSClient now, but I'll do that in the next PR